dot/parachain/network-bridge/receiver.go

```diff
@@ -5,33 +5,218 @@ package networkbridge
+	perRelayParent map[common.Hash]PerRelayParent // map[relay parent]PerRelayParent
+
+	// Collations that we have successfully requested from peers and waiting
+	// on validation.
+	fetchedCandidates map[string]CollationEvent
+	// heads are sorted in descending order by block number
+	liveHeads []parachaintypes.ActivatedLeaf
+
+	finalizedNumber uint32
+
+	OverseerToSubSystem  <-chan any
+	SubsystemsToOverseer chan<- any
+
+	networkEventInfoChan chan *network.NetworkEventInfo
+
+	authorityDiscoveryService AuthorityDiscoveryService
+}
+
+type CollationStatus int
+
+const (
+	// We are waiting for a collation to be advertised to us.
+	Waiting CollationStatus = iota
+	// We are currently fetching a collation.
+	Fetching
+	// We are waiting that a collation is being validated.
+	WaitingOnValidation
+	// We have seconded a collation.
+	Seconded
+)
+
+type CollationEvent struct {
+	CollatorId       parachaintypes.CollatorID
+	PendingCollation PendingCollation
+}
+
+// Identifier of a fetched collation
+type fetchedCollationInfo struct {
+	// Candidate's relay parent
+	relayParent   common.Hash
+	paraID        parachaintypes.ParaID
+	candidateHash parachaintypes.CandidateHash
+	// Id of the collator the collation was fetched from
+	collatorID parachaintypes.CollatorID
+}
+
+func (f fetchedCollationInfo) String() string {
+	return fmt.Sprintf("relay parent: %s, para id: %d, candidate hash: %s, collator id: %+v",
+		f.relayParent.String(), f.paraID, f.candidateHash.Value.String(), f.collatorID)
+}
+
+func fetchedCandidateFromString(str string) (fetchedCollationInfo, error) {
+	splits := strings.Split(str, ",")
+	if len(splits) != 4 {
+		return fetchedCollationInfo{}, fmt.Errorf("%w: %s", ErrInvalidStringFormat, str)
+	}
+
+	relayParent, err := common.HexToHash(strings.TrimSpace(splits[0]))
+	if err != nil {
+		return fetchedCollationInfo{}, fmt.Errorf("getting relay parent: %w", err)
+	}
+
+	paraID, err := strconv.ParseUint(strings.TrimSpace(splits[1]), 10, 64)
+	if err != nil {
+		return fetchedCollationInfo{}, fmt.Errorf("getting para id: %w", err)
+	}
+
+	candidateHashBytes, err := common.HexToBytes(strings.TrimSpace(splits[2]))
+	if err != nil {
+		return fetchedCollationInfo{}, fmt.Errorf("getting candidate hash bytes: %w", err)
+	}
+
+	candidateHash := parachaintypes.CandidateHash{
+		Value: common.NewHash(candidateHashBytes),
+	}
+
+	var collatorID parachaintypes.CollatorID
+	collatorIDBytes, err := common.HexToBytes(strings.TrimSpace(splits[3]))
+	if err != nil {
+		return fetchedCollationInfo{}, fmt.Errorf("getting collator id bytes: %w", err)
+	}
+	copy(collatorID[:], collatorIDBytes)
+
+	return fetchedCollationInfo{
+		relayParent:   relayParent,
+		paraID:        parachaintypes.ParaID(paraID),
+		candidateHash: candidateHash,
+		collatorID:    collatorID,
+	}, nil
+}
+
+type PerRelayParent struct {
+	prospectiveParachainMode parachaintypes.ProspectiveParachainsMode
+	assignment               *parachaintypes.ParaID
+}
```

Comment:
The struct is exported, but not the fields. Is this intentional?

Reply:
fields are not getting used now.